src/Shared/RoslynUtils/ParsabilityHelper.cs

```diff
@@ -185,13 +173,13 @@ internal static Bindability GetBindability(ITypeSymbol typeSymbol, WellKnownType
                     if (methodSymbolCandidate is IMethodSymbol methodSymbol)
                     {
                         bindAsyncMethod ??= methodSymbol;
-                        if (IsBindAsyncWithParameter(methodSymbol, typeSymbol, wellKnownTypes))
+                        if (IsBindAsyncWithParameter(methodSymbol, typeSymbol))
                         {
                             bindabilityMethod = BindabilityMethod.BindAsyncWithParameter;
                             bindMethodSymbol = methodSymbol;
                             break;
                         }
-                        if (IsBindAsync(methodSymbol, typeSymbol, wellKnownTypes))
+                        if (IsBindAsync(methodSymbol, typeSymbol))
```

Comment:
probably not worth the bother, but IsBindAsync and IsBindAsyncWithParameter are similar and could be one method that has an out bool returning whether there's a parameter.